clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/ops/DestroyGoogleServerGroupAtomicOperation.groovy

```diff
@@ -88,34 +136,31 @@ class DestroyGoogleServerGroupAtomicOperation implements AtomicOperation<Void> {
 
         // We must make sure the autoscaler is deleted before deleting the managed instance group.
         googleOperationPoller.waitForZonalOperation(compute, project, zone, autoscalerDeleteOperationName, null, task,
-            "zonal autoscaler $serverGroupName", BASE_PHASE)
+          "zonal autoscaler $serverGroupName", BASE_PHASE)
       }
+      null
     }
+  }
 
-    def instanceGroupManagerDeleteOperation =
-        isRegional
-        ? compute.regionInstanceGroupManagers().delete(project, region, serverGroupName).execute()
-        : compute.instanceGroupManagers().delete(project, zone, serverGroupName).execute()
-    def instanceGroupOperationName = instanceGroupManagerDeleteOperation.getName()
+  Closure destroyInstanceGroup(Compute compute, String serverGroupName, String project, String region, String zone, Boolean isRegional) {
+    return {
+      def instanceGroupManagerDeleteOperation = isRegional ?
+        compute.regionInstanceGroupManagers().delete(project, region, serverGroupName).execute() :
+        compute.instanceGroupManagers().delete(project, zone, serverGroupName).execute()
 
-    task.updateStatus BASE_PHASE, "Waiting on delete operation for managed instance group..."
+      def instanceGroupOperationName = instanceGroupManagerDeleteOperation.getName()
 
-    // We must make sure the managed instance group is deleted before deleting the instance template.
-    if (isRegional) {
-      googleOperationPoller.waitForRegionalOperation(compute, project, region, instanceGroupOperationName, null, task,
+      task.updateStatus BASE_PHASE, "Waiting on delete operation for managed instance group..."
+
+      // We must make sure the managed instance group is deleted before deleting the instance template.
+      if (isRegional) {
+        googleOperationPoller.waitForRegionalOperation(compute, project, region, instanceGroupOperationName, null, task,
           "regional instance group $serverGroupName", BASE_PHASE)
-    } else {
-      googleOperationPoller.waitForZonalOperation(compute, project, zone, instanceGroupOperationName, null, task,
+      } else {
+        googleOperationPoller.waitForZonalOperation(compute, project, zone, instanceGroupOperationName, null, task,
           "zonal instance group $serverGroupName", BASE_PHASE)
+      }
+      null
```

Comment:
Groovy tries to return the `Operation` result coming from either of the `googleOperationPoller`s otherwise. The original code does this too for similar reasons [here](https://github.com/spinnaker/clouddriver/pull/740/files/d2302fbebddb0ac750cbd111d85d57f9b4d3d3e9#diff-c74180f4c1b005c9f9b2b0453de7fa5dL119).